packages/toc/src/generators/markdown/index.ts

```diff
@@ -66,19 +80,31 @@ function generate(editor: IDocumentWidget<FileEditor>): INumberedHeading[] {
  * @param tracker - file editor tracker
  * @param widget - table of contents widget
  * @param sanitizer - HTML sanitizer
+ * @param settings - advanced settings for toc extension
  * @returns ToC generator capable of parsing Markdown files
  */
 function createMarkdownGenerator(
   tracker: IEditorTracker,
   widget: TableOfContents,
   sanitizer: ISanitizer,
-  translator?: ITranslator
+  translator?: ITranslator,
+  settings?: ISettingRegistry.ISettings
 ): Registry.IGenerator<IDocumentWidget<FileEditor>> {
-  const options = new OptionsManager(widget, {
+  let numberingH1 = true;
+  if (settings) {
+    numberingH1 = settings.composite.numberingH1 as boolean;
+  }
+  let options = new OptionsManager(widget, {
```

Comment:
```suggestion
  const options = new OptionsManager(widget, {
```
`options` is still constant as you modify an attribute of the object not the object itself.